### PR TITLE
Events should be queried from BSC in block ranges of no more than 5000 blocks

### PR DESCRIPTION
**Pull request by redben**
This issue only shows up on BSC Mainnet.

BSC Events Block Range

### CODE REVIEW

**Comment by redben** [comment]
telegram-cloud-photo-size-2-5208678746381464323-x

File: src/store/gateway/ethereum.ts [hunk 1]
```diff
@@ -538,11 +538,16 @@ export async function refreshEthereumHistory(context: ActionContext) {
...
 async function logEvents(address, gateway, symbol, depositEvent, withdrawEvent) {
   const cached = ethereumModule.state.history
-  const fromBlock = cached.length ? cached[0].blockNumber : 0
-  const range = {
-    fromBlock,
-    toBlock: "latest",
+  let fromBlock = cached.length ? cached[0].blockNumber : 0
+  let toBlock: string | number = "latest"
+  // BSC max query size is 5000
+  if (ethereumModule.state.walletType === "binance") {
```
Comment (enlight) [inline src/store/gateway/ethereum.ts:544]: Should work based on the network id/name rather than the wallet type

Comment (redben) [reply src/store/gateway/ethereum.ts]: changed to `ethereumModule.state.networkName.startsWith("bsc-")`

**Review by enlight — COMMENTED** [review]
LGTM

**Comment by yuneeksehgal** [comment]
History page is showing the Binance Smart Chain Txs

image

**Pull request merged**